services/gitlab/gitlab-base.js

```diff
@@ -0,0 +1,19 @@
+import { BaseJsonService } from '../index.js'
+
+export default class GitLabBase extends BaseJsonService {
+  static auth = {
+    passKey: 'gitlab_token',
+    serviceKey: 'gitlab',
```

Comment:
We'll constrain the allowed origins for Shields.io to `gitlab.com`, but need this to be flexible for other self-hosters which is why the key is used here instead of hardcoding the `authorizedOrigins` value to `gitlab.com`